src/state/apis/swappers/selectors.ts

```diff
@@ -0,0 +1,7 @@
+import { QueryStatus } from '@reduxjs/toolkit/query'
+import type { ReduxState } from 'state/reducer'
+
+export const selectSwappersApiTradeQuotePending = (state: ReduxState) =>
+  Object.values(state.swappersApi.queries).some(
+    query => query?.endpointName === 'getTradeQuote' && query?.status === QueryStatus.pending,
```

Comment:
This will return true if there is a stale query still pending while the current query is completed. 

There is 1 source of truth here, so multiple instances of the `useGetTradeQuote` shouldnt cause us dramas, but in any case invoking the query requires use of the hook so it would make more sense to tightly couple the concept of loading with instances of the hook, i.e

`const { isPending } = useGetTradeQuote()`

This way you wont have to engineer the selection of loading state with race condition across various requests and parts of the app using the same hook

Reply:
Alternatively you can include logic here to check the query start time like you did below

```
if ( !state.lastRequestStartTime || action.payload.queryStartTime > state.lastRequestStartTime) {...}
```

Reply:
something like

```
return Object.values(state.swappersApi.queries).some(
  query => query && 
    query.endpointName === 'getTradeQuote' &&
    query.status === QueryStatus.pending &&
    (!state.lastRequestStartTime || latestPendingQuery.queryStartTime > state.lastRequestStartTime)
)
```

Reply:
Good point. I opted for a more explicit version of your selector, which checks only if the most recently started query is pending (in case there are multiple new queries in flight before one returns - a race condition of a race condition).

See https://github.com/shapeshift/web/pull/4847/commits/c74339b12b9b75e9f5930866150318945d88049c